pandas/core/ops/array_ops.py

```diff
@@ -533,7 +538,11 @@ def maybe_prepare_scalar_for_op(obj, shape: Shape):
             from pandas.core.arrays import DatetimeArray
 
             # Avoid possible ambiguities with pd.NaT
-            obj = obj.astype("datetime64[ns]")
+            # GH 52295
+            if is_unitless(obj.dtype) or not is_supported_unit(
```

Comment:
i think in non-supported unit cases we'd want to use the closest-to-supported?  could just handle the unitless case here?

Reply:
Good call. We also need to handle the unsupported NaT case here too though